src/main/java/uk/ac/ebi/embl/converter/gff3/reader/GFF3FileReader.java

```diff
@@ -44,6 +44,7 @@ public class GFF3FileReader implements AutoCloseable {
     GFF3Annotation currentAnnotation;
     String currentAccession;
     Map<String, GFF3SequenceRegion> accessionSequenceRegionMap = new HashMap<>();
+    HashSet<String> finishedAccessions = new HashSet<>();
```

Comment:
Suggestion to inject this HashSet to GFF3FileReader as follows
```
 // Set of converted accessions.
 HashSet<String> convertedAccessions = new HashSet<>();
  try (GFF3FileReader gff3Reader = new **GFF3FileReader(reader,convertedAccessions))** {
```
And then add it to the Set after returning from the readAnnotation()

while ((currentAnnotation = gff3Reader.readAnnotation()) != null) {
    ...
    if (isSameAnnotation(previousAnnotation, currentAnnotation)) {
        previousAnnotation.merge(currentAnnotation);
    } else {
        ....
        **convertedAccessions.add(previousAnnotation.getAccession());**
    }
}

Reply:
but it breaks the encapsulation of the reader, now the callers needs to be aware of its internal logic.